services/app-web/src/components/ChangeHistory/ChangeHistoryV2.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react'
+import { dayjs } from '../../common-code/dateHelpers/dayjs'
+import { SectionHeader } from '../SectionHeader'
+import { Accordion, Link } from '@trussworks/react-uswds'
+import type { AccordionItemProps } from '@trussworks/react-uswds/lib/components/Accordion/Accordion'
+import { UpdateInformation, Contract } from '../../gen/gqlClient'
+import styles from './ChangeHistory.module.scss'
+type ChangeHistoryProps = {
+    contract: Contract
+}
+
+type flatRevisions = UpdateInformation & {
+    kind: 'submit' | 'unlock'
+    revisionVersion: string | undefined
+}
+
+export const ChangeHistoryV2 = ({
+    contract,
+}: ChangeHistoryProps): React.ReactElement => {
+    const flattenedRevisions = (): flatRevisions[] => {
+        const result: flatRevisions[] = []
+        //Reverse revisions to order from earliest to latest revision. This is to correctly set version for each
+        // contract & recontract.
+        const reversedRevisions = [...contract.packageSubmissions].reverse()
+        reversedRevisions.forEach((r, index) => {
+            if (r.contractRevision.unlockInfo) {
+                const newUnlock: flatRevisions = {} as flatRevisions
+                newUnlock.updatedAt = r.contractRevision.unlockInfo.updatedAt
+                newUnlock.updatedBy = r.contractRevision.unlockInfo.updatedBy
+                newUnlock.updatedReason =
+                    r.contractRevision.unlockInfo.updatedReason
+                newUnlock.kind = 'unlock'
+                //Use unshift to push the latest revision unlock info to the beginning of the array
+                result.unshift(newUnlock)
+            }
+            if (r.contractRevision.submitInfo) {
```

Comment:
this is necessary b/c it is optional in the types, but just to reiterate every revision in packageSubmissions must have this set.